lib/MTDropbox.js

```diff
@@ -0,0 +1,166 @@
+      dbx({
+          resource: 'files/list_folder',
+          parameters: getDbxListFileFromPathParam(path)
+      }, async function(err, result, response) {
+          if (err) { return reject(err); }
+          // DEBUG // console.log(result); console.log(response.headers);
+          const initialBackupsCount = result.length;
+          const deprecatedBackups = result.entries
+                                  .filter(e => e[".tag"] === "file")
+                                  .filter(e => new Date(e.client_modified) < new Date(ctimeMsMax))
+                                  .map(e=> { return {
+                                      name: e.name,
+                                      path_lower: e.path_lower,
+                                      client_modified: e.client_modified
+                                    };
+                                  });
+          const deprecatedBackupsCount = deprecatedBackups.length;
+          const deletedBackups = await mt.backupsToClean(dbx, dryMode, deprecatedBackups, cleanCount, minCount);
+          const cleanedCount = deletedBackups.length;
+          const cleanedFiles = deletedBackups.map(db => db.path_lower);
+          // DEBUG //  console.log('fileNames', fileNames)
+          return resolve({initialBackupsCount, deprecatedBackupsCount, cleanedCount, cleanedFiles});
+      });
+    });
+  }
+
+  backupsToClean(dbx, dryMode, deprecatedBackups, cleanCount, minCount) {
+    if (deprecatedBackups === null || deprecatedBackups === undefined || deprecatedBackups.length <= minCount) {
+      return [];
+    }
+    // sort by client_modified asc
+    deprecatedBackups = deprecatedBackups.sort((a, b)=>{
+       return (a.client_modified > b.client_modified) - (a.client_modified < b.client_modified);// client_modified asc
+    });
+    // DEBUG // console.log("dbx backupsToClean", {deprecatedBackups, cleanCount, minCount});
+    // keep nb to clean
+    var toDelete = deprecatedBackups.slice(0, Math.max(0, deprecatedBackups.length + 1 - minCount));
+    // DEBUG // console.log("toDelete 1/2", {toDelete});
+    toDelete = toDelete.slice(0, Math.min(cleanCount, toDelete.length));
+    // DEBUG // console.log("toDelete 2/2", {toDelete});
+
+    for (const toDeleteEntry of toDelete) {
+      if (!dryMode) {
+        dbx({
+            resource: 'files/delete',
+            parameters: {path: toDeleteEntry.path_lower}
+        }, async function(err, result, response) {
+            if (err) { return Promise.reject(err); }
+            // DEBUG // console.log(result); console.log(response.headers);
+        });
+      } else {
+        console.log("*dry mode* DELETE", toDeleteEntry.path_lower )
+      }
+    }
+    return Promise.resolve(toDelete);
+  }
+}
+
+//~ private
+
+//https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder
+function getDbxListFileFromPathParam(path) {
+  return {path, "recursive":false, limit: 2000, include_non_downloadable_files: false};

```

Comment:
because of limit 2000 nd without pagination handler, there is a need to prevent rotation feature when there is more result and warn user